src/vm/functions/mod.rs

```diff
@@ -542,7 +553,7 @@ fn special_let(
                     return Err(CheckErrors::NameAlreadyUsed(binding_name.clone().into()).into())
                 }
 
-            let binding_value = eval(var_sexp, env, context)?;
+            let binding_value = eval(var_sexp, env, &inner_context)?;
```

Comment:
Will this let me re-bind the same variable, like so?

```
(let (
   (x 0)
   (x (+ x 1))
   (x (+ x 2))
) (
   (print x) ;; should print 3
))
```

It's fine by me if it doesn't; I'm just curious.

Reply:
No, it won't -- that's a type of variable shadowing, which we don't allow in Clarity.